core/src/main/java/com/graphhopper/storage/GraphEdgeIdFinder.java

```diff
@@ -1,240 +0,0 @@
-     * @param useEdgeIdsUntilAreaSize until the specified area (specified in m²) use the findEdgesInShape method
-     */
-    public BlockArea parseBlockArea(String blockAreaString, EdgeFilter filter, double useEdgeIdsUntilAreaSize) {
-        final String objectSeparator = ";";
-        final String innerObjSep = ",";
-        BlockArea blockArea = new BlockArea(graph);
-
-        // Add blocked circular areas or points
-        if (!blockAreaString.isEmpty()) {
-            String[] blockedCircularAreasArr = blockAreaString.split(objectSeparator);
-            for (int i = 0; i < blockedCircularAreasArr.length; i++) {
-                String objectAsString = blockedCircularAreasArr[i];
-                String[] splittedObject = objectAsString.split(innerObjSep);
-
-                Shape shape;
-                boolean point = false;
-                // always add the shape as we'll need this for virtual edges and for debugging.
-                if (splittedObject.length > 4) {
-                    shape = Polygon.parsePoints(objectAsString);
-                } else if (splittedObject.length == 4) {
-                    final BBox bbox = BBox.parseTwoPoints(objectAsString);
-                    final RectangleLineIntersector cachedIntersector = new RectangleLineIntersector(toEnvelope(bbox));
-                    shape = new BBox(bbox.minLon, bbox.maxLon, bbox.minLat, bbox.maxLat) {
-                        @Override
-                        public boolean intersects(PointList pointList) {
-                            return BBox.intersects(cachedIntersector, pointList);
-                        }
-                    };
-                } else if (splittedObject.length == 3) {
-                    double lat = Double.parseDouble(splittedObject[0]);
-                    double lon = Double.parseDouble(splittedObject[1]);
-                    int radius = Integer.parseInt(splittedObject[2]);
-                    shape = new Circle(lat, lon, radius);
-                } else if (splittedObject.length == 2) {
-                    double lat = Double.parseDouble(splittedObject[0]);
-                    double lon = Double.parseDouble(splittedObject[1]);
-                    shape = new Circle(lat, lon, P_RADIUS);
-                    point = true;
-                } else {
-                    throw new IllegalArgumentException(objectAsString + " at index " + i + " need to be defined as lat,lon "
-                            + "or as a circle lat,lon,radius or rectangular lat1,lon1,lat2,lon2");
-                }
-                
-                
-                if (point || calculateArea(shape) <= useEdgeIdsUntilAreaSize) {
-                    GHIntHashSet blockedEdges = findEdgesInShape(shape, filter);
-                    if (!blockedEdges.isEmpty()) {
-                        blockArea.add(shape, blockedEdges);
-                    }
-                } else {
-                    blockArea.add(shape);
-                }
-            }
```

Comment:
With this optimization we used a hash set of blocked / affected edges instead of point-in-area checks for small areas. Not sure if we could/should do the same for (small) custom model areas?

Reply:
I vaguely remember @karussell mentioning that this feature doesn't work well with virtual edges, but maybe I'm completely wrong.

Nevertheless , we should aim to improve the performance as noted in #2313. Wasting 30% of execution time and 80% of memory allocation just for bbox checking is pretty nasty.

There's also #2625 which aims to improve precision. Maybe we should start with that and see if it improves performance as well?

Reply:
> And getting rid of legacy should get a higher priority than an optimization.

Absolutely. That would be out of scope for this PR.

Reply:
Ok, so let's just keep in mind that there was this optimization and maybe come back to this later.